streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -97,31 +96,27 @@ public void commit() {
     }
 
     @Override
-    public void close() {
-        //no-op
+    public void suspend() {
+        commit();
     }
 
     @Override
-    public void initTopology() {
-        //no-op
+    public void close() {
+        try {
+            commit();
+            closeStateManager(true);
+        } catch (final RuntimeException e) {
```

Comment:
Or is your concern about `commit` throwing an exception? If that is the case, then the `try`/`catch` should just be around that method IMO.